### PR TITLE
Fix yolov4 example bgr2rgb

### PR DESCRIPTION
**Pull request by MarsTechHAN**

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=2051) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=2051) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=2051) it.</sub>

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -7,9 +7,9 @@ macro(ncnn_add_example name)
     set_property(TARGET ${name} PROPERTY FOLDER "examples")
 endmacro()
 
-find_package(OpenCV QUIET COMPONENTS core highgui imgproc imgcodecs)
+find_package(OpenCV QUIET COMPONENTS core highgui imgproc imgcodecs videoio)
 if(NOT OpenCV_FOUND)
-    find_package(OpenCV QUIET COMPONENTS core highgui imgproc)
+    find_package(OpenCV QUIET COMPONENTS core highgui imgproc videoio)
```
Comment (nihui) [inline examples/CMakeLists.txt:12]: This line is for opencv 2.x, videoio is not needed as highgui is enough

File: examples/yolov4.cpp [hunk 2]
```diff
@@ -15,12 +15,22 @@
 #include "net.h"
 
 #include <opencv2/core/core.hpp>
+#include <opencv2/videoio/videoio.hpp>
```
Comment (nihui) [inline examples/yolov4.cpp:18]: There is no videoio in opencv 2.x
guard with version macro
```
CV_MAJOR_VERSION
CV_MINOR_VERSION
```

File: examples/yolov4.cpp [hunk 3]
```diff
@@ -15,12 +15,22 @@
...
+
+#include <string.h>
+#include <stdio.h>
 
-#define YOLOV4_TINY 1 //0 or undef for yolov4
+#define NCNN_PROFILING
+#define YOLOV4_TINY //Using yolov4_tiny, if undef, using original yolov4
+
+#ifdef NCNN_PROFILING
+#include <chrono>
```
Comment (nihui) [inline examples/yolov4.cpp:32]: ncnn benchmark.h has the necessary timestamp functions
Besides, c++11 is not mandatory when building with opencv 2.x

File: examples/yolov4.cpp [hunk 4]
```diff
@@ -29,42 +39,72 @@ struct Object
...
-static int detect_yolov4(const cv::Mat& bgr, std::vector<Object>& objects)
+static int init_yolov4(ncnn::Net* yolov4, int* target_size)
 {
-    ncnn::Net yolov4;
+    /* --> Set the params you need for the ncnn inference <-- */
+
+    yolov4->opt.num_threads = 4; //You need to compile with libgomp for multi thread support
 
-    yolov4.opt.use_vulkan_compute = true;
+    yolov4->opt.use_vulkan_compute = true; ////You need to compile with libvulkan for multi thread support
```
Comment (nihui) [inline examples/yolov4.cpp:48]: ////You need to compile with libvulkan for gpu support

File: examples/yolov4.cpp [hunk 5]
```diff
@@ -136,30 +176,139 @@ static void draw_objects(const cv::Mat& bgr, const std::vector<Object>& objects)
...
+                fprintf(stderr, "OpenCV Failed to Capture from device %s\n", devicepath);
+                return -1;
+            }
+        }
+
+#ifdef NCNN_PROFILING
+        auto t_detect_start = std::chrono::high_resolution_clock::now();
+#endif
+
+        detect_yolov4(frame, objects, target_size, &yolov4); //Creat an extractor and run detection
```
Comment (nihui) [inline examples/yolov4.cpp:287]: // Create

**Review by nihui — CHANGES_REQUESTED** [review]

**Review by nihui — APPROVED** [review]

**Comment by nihui** [comment]
hanxiao yes !

**Pull request merged**